MediaBrowser.Api/Playback/Hls/HLSCodecStringFactory.cs

```diff
@@ -0,0 +1,152 @@
+            result.Append("mp4a");
+            if (string.Equals(profile, "HE", StringComparison.InvariantCultureIgnoreCase))
+            {
+                result.Append(".40.5");
+            } else
+            {
+                //default to LC if profile is invalid
+                result.Append(".40.2");
+            }
+
+            return result.ToString();
+        }
+
+        /// <summary>
+        /// Gets a H.264 codec string
+        /// </summary>
+        /// <param name="profile">H.264 profile</param>
+        /// <param name="level">H.264 level</param>
+        /// <returns>H.264 string</returns>
+        public static string GetH264String(string profile, int level)
+        {
+            StringBuilder result = new StringBuilder((int) MaxVideoStringLength.H264);
+            string levelHex = level.ToString("X2");
+            result.Append("avc1");
+
+            if (string.Equals(profile, "baseline", StringComparison.InvariantCultureIgnoreCase))
+            {
+                result.Append(".42E0");
+            } else if (string.Equals(profile, "main", StringComparison.InvariantCultureIgnoreCase))
+            {
+                result.Append(".4D40");
+            } else if (string.Equals(profile, "high", StringComparison.InvariantCultureIgnoreCase))
+            {
+                result.Append(".6400");
+            } else
+            {
+                //default to constrained baseline if profile is invalid
+                result.Append(".4240");
+            }
+
+            result.Append(levelHex);
+            return result.ToString();
+        }
+
+        /// <summary>
+        /// Gets a H.265 codec string
+        /// </summary>
+        /// <param name="profile">H.265 profile</param>
+        /// <param name="level">H.265 level</param>
+        /// <returns>H.265 string</returns>
+        public static string GetH265String(string profile, int level)
+        {
+            //The h265 syntax is a bit of a mystery at the time this comment was written.
+            //This is what I've found through various sources:
+            //FORMAT: [codecTag].[profile].[constraint?].L[level * 30].[UNKNOWN]
+            StringBuilder result = new StringBuilder((int) MaxVideoStringLength.H265);
+
+            result.Append("hev1");
+            if (string.Equals(profile, "main10", StringComparison.InvariantCultureIgnoreCase))
+            {
+                result.Append(".2.6");
+            } else
+            {
+                //default to main if profile is invalid
+                result.Append(".1.6");
+            }
+
+            result.Append(".L" + (level * 3));
+            result.Append(".B0");
+            return result.ToString();
+        }
+
+        /// <summary>
+        /// Gets an AC-3 codec string
+        /// </summary>
+        /// <returns>AC-3 codec string</returns>
+        public static string GetAC3String()
+        {
+            string result = "mp4a.a5";
```

Comment:
No need to assign it to a variable first

Reply:
Oh right, that's a bad habit of mine 😬